runtime/tests/demangle.c

```diff
@@ -0,0 +1,93 @@
+    "_B_foo_bar_baz",                                       // foo_bar_baz
+    "_B_foo23",                                             // foo23
+    "_Bb1x6choreohello",                                    // ballerinax/choreo.hello
+    "_Bb0m4lang5valuetoString",                             // ballerina/lang.value.toString
+    "_B4wso2m10nballerinam5front6syntaxparse",              // wso2/nballerina.front.syntax.parse
+    "_B0m7foo_bar3bazhello",                                // foo_bar.baz.hello
+    "_Bb1__2ioprintln",                                     // ballerina_/io.println
+    "_Bb1_22ioprintln",                                     // ballerina2/io.println
+    "_Bb2_232ioprintln",                                    // ballerina23/io.println
+    "_Bb10_1234_x12342ioprintln",                           // ballerina1234_x1234/io.println
+    "_Bb10_1234_x1234m10nballerinam5front6syntaxparse",     // ballerina1234_x1234/nballerina.front.syntax.parse
+    "_B10_1234_x1234m10nballerinam5front6syntaxparse",      // 1234_x1234/nballerina.front.syntax.parse
+    "_B1__m10nballerinam5front6syntaxparse",                // _/nballerina.front.syntax.parse
+    "_Bb0m1__2ioprintln",                                   // ballerina/_.io.println
+    "_Bb0m1_22ioprintln",                                   // ballerina/2.io.println
+    "_Bb0m7foo_bar2ioprintln",                              // ballerina/foo_bar.io.println
+    "_Bb0m7foo_barm4__bar2ioprintln",                       // ballerina/foo_bar._bar.io:println
+    "_Bb0m7foo_barm4_2bar2ioprintln",                       // ballerina/foo_bar.2bar.io:println
+    "_Bb0m7foo_barm4_2bar3__ioprintln",                     // ballerina/foo_bar.2bar._io:println
+    "_Bb0m7foo_barm4_2bar2io_println",                      // ballerina/foo_bar.2bar.io:_println
+    "_Bb0m7foo_barm4_2bar2io_3println",                     // ballerina/foo_bar.2bar.io:_println
+    "_Bb0m7foo_barm4_2bar21new_nballerina_module_3println", // ballerina/foo_bar.2bar.new_nballerina_module:_3println
+    "_Bb0m7foo_barm4_2bar20new_ballerina_module_3println"   // ballerina/foo_bar.2bar.new_ballerina_module:_3println
+};
+
+const char expectedOutput[NAMES_COUNT][50] = {
+    "io:println\n",
+    "foo\n",
+    "foo_bar_baz\n",
+    "foo23\n",
+    "choreo:hello\n",
+    "lang.value:toString\n",
+    "nballerina.front.syntax:parse\n",
+    "foo_bar.baz:hello\n",
+    "io:println\n",
+    "io:println\n",
+    "io:println\n",
+    "io:println\n",
+    "nballerina.front.syntax:parse\n",
+    "nballerina.front.syntax:parse\n",
+    "nballerina.front.syntax:parse\n",
+    "_.io:println\n",
+    "2.io:println\n",
+    "foo_bar.io:println\n",
+    "foo_bar._bar.io:println\n",
+    "foo_bar.2bar.io:println\n",
+    "foo_bar.2bar._io:println\n",
+    "foo_bar.2bar.io:_println\n",
+    "foo_bar.2bar.io:_3println\n",
+    "foo_bar.2bar.new_nballerina_module:_3println\n",
+    "foo_bar.2bar.new_ballerina_module:_3println\n"
```

Comment:
Is it possible to avoid `\n`on each case, and to just check it in the test?

Reply:
Yes, we can do that. The reason to use '\n' is to indicate the end of the demangled name.

Reply:
C strings do end in `\0` automatically, so why do we need another maker ?

Reply:
Yes, that is correct. When printing to an output file, I am adding a new line. That is why I am checking the new line. But removed the new line and checking the end of line in the test now.
Fixed